src/TestProblem/Hydro/CCSN/Flag_CCSN.cpp

```diff
@@ -34,43 +41,43 @@ extern double CCSN_CentralDens;
 bool Flag_CoreCollapse( const int i, const int j, const int k, const int lv, const int PID, const double *Threshold )
 {
 
-   bool Flag = false;
+   bool Flag      = false;
    bool MaxRefine = false;
 
-   const double dh        = amr->dh[lv];
-   const double Center[3] = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
-   const double Pos   [3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,
-                              amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
-                              amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
-
-   const double dx = Center[0] - Pos[0];
-   const double dy = Center[1] - Pos[1];
-   const double dz = Center[2] - Pos[2];
-   const double r  = sqrt(  SQR( dx ) + SQR( dy ) + SQR( dz )  );
+   const double dh     = amr->dh[lv];
+   const double Pos[3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,
+                           amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
+                           amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
+#  ifdef GRAVITY
+   const double dR [3] = { Pos[0]-GREP_Center[0], Pos[1]-GREP_Center[1], Pos[2]-GREP_Center[2] };
+#  else
+   const double dR [3] = { NULL };
```

Comment:
I prefer to define `Center = amr->BoxCenter` or `GREP_Center` according to `GRAVITY`. If you want to modify `dR` instead, set it to `Pos - amr->BoxCenter` when `GRAVITY` is disabled.

Reply:
I have updated this.